### ISSUE TITLE
AppData call takes longer than 5 seconds

### ISSUE THREAD

**Issue by Orscheler** [RainLoop/rainloop-webmail#2161]
**RainLoop version, browser, OS:**
1.16.0 / Chrome & Curl 

**Expected behavior and actual behavior:**
/?/AppData call takes longer than 5 seconds both on my local installation, as well as in the rainloop-demo 

**Steps to reproduce the problem:**
When loading Rainloop or when changing between multiple accounts request the mentioned call is processed and slows down the usage

**Logs or screenshots:**
![appdata](https://user-images.githubusercontent.com/37636820/151680252-9844e39b-d7f5-400a-9df4-84fa287b86ea.png)

**Comment by Furrypaw** [comment]
Yes I also observe this on both of my servers. Seems to even happen to official demo page.
Somehow I assume it could be related to Subscription server on their side. As if you go to Admin Panel -> Licensing
![ZO7xJZ](https://user-images.githubusercontent.com/71909410/151684919-ac369cee-e983-4a90-9531-98e5cd08ad58.png)
(Keep in mind that this is my wild guess)

**Comment by yeupou** [comment]
it is possible it is actually the same as issue #2160 - on login page, for some reason, there is a curl to http://api.rainloop.net/status/FQDN 

as api.rainloop.net seems dead, it incurs a timeout.

The question is why api.rainloop.net is even called at every login. I was not aware, when installing the server that it would make such call to a distant server. 
I am not sure to understand clearly the potential privacy issues here.

**Comment by Furrypaw** [comment]
As a temporary fix I migrated over to community edition, it seems to work fine. As I did not have license im not sure if your branding carries over or not.

**Comment by rainloopsay919** [comment]
> it is possible it is actually the same as issue #2160 - on login page, for some reason, there is a curl to http://api.rainloop.net/status/FQDN
> 
> as api.rainloop.net seems dead, it incurs a timeout.
> 
> The question is why api.rainloop.net is even called at every login. I was not aware, when installing the server that it would make such call to a distant server. I am not sure to understand clearly the potential privacy issues here.

I can confirm it is the same issue.

These requests are related to subscription status verification:

- `http://api.rainloop.net/status/FQDN` - checks the subscription status for _FQDN_. Expects one of the following responses:
  - _NO_ - no subscription found
  - _EXPIRED:timestamp_ - subscription found, valid until _timestamp_
  - _EXPIRED:1898625600_ or _EXPIRED:1898625700_ - a special case of the previous response, displayed as unlimited subscription
- `http://api.rainloop.net/activate/FQDN/key` - attempts to activate a subscription for _FQDN_ using _key_.
  - _OK_ - assumed successful activation
  - _TOO_MANY_CONNECTIONS_ - assumed failed activation due to rate limit
  - _ERROR:string_ - assumed failed activation, the reason is specified in _string_

The first request is sent when accessing both user and admin interface in order to disable premium features when the subscription is expired or had never existed for the _FQDN_. The latter is sent when subscription key activation is attempted (after passing local checks).

The API is not used for anything else at the moment.

> As a temporary fix I migrated over to community edition, it seems to work fine. As I did not have license im not sure if your branding carries over or not.

Or you can use #2148, as the legal status of this code is uncertain, both because of the presence of AGPLv3 LICENSE file in the root of this repository, and #450. Alternatively, switch to something which doesn't have uncertain support status for years.

**Comment by protonwill390** [comment]
A thing was missed, _TOO_MANY_CONNECTIONS_ exists for the first request as well..

**Comment by rainloopano43** [comment]
@yeupou I just realized I did not explain why it is the same issue as #2160, at all.

As you can see [here](https://github.com/RainLoop/rainloop-webmail/blob/master/rainloop/v/0.0.0/app/libraries/RainLoop/Providers/Prem.php#L67), this function is in Prem provider, and it calls `this->Type()`.

`Type` function calls `$this->Parser($this->Fetcher(false, true))` and gets its result, as you can see [here](https://github.com/RainLoop/rainloop-webmail/blob/master/rainloop/v/0.0.0/app/libraries/RainLoop/Providers/Prem.php#L25).

In the `Fetcher` function, you can see the request itself [here](https://github.com/RainLoop/rainloop-webmail/blob/master/rainloop/v/0.0.0/app/libraries/RainLoop/Providers/Prem.php#L192-L196).

[Here](https://github.com/RainLoop/rainloop-webmail/blob/master/rainloop/v/0.0.0/include.php#L44) you can see the line which defines `APP_API_PATH`.

So it takes some time until the request times out, and the execution proceeds.

**Comment by Pofilo** [comment]
> As a temporary fix I migrated over to community edition, it seems to work fine. As I did not have license im not sure if your branding carries over or not.

How did you manage to do that ?
I do not have a license too but I have the timeout as well.

**Comment by yeupou** [comment]
In my case, I decided it was easier to move to snappymail.

Plus, I did not like much finding by accident that servers where pinging a distant server while I never subscribed to anything like this and found nothing in this regards in config.

**Comment by Furrypaw** [comment]
> > As a temporary fix I migrated over to community edition, it seems to work fine. As I did not have license im not sure if your branding carries over or not.
> 
> How did you manage to do that ? I do not have a license too but I have the timeout as well.

In my case:
-Download community edition from official site
-In previous instalation folder remove everything but the data folder (remember to keep backups of everything)
-Extract index.php and rainloop folder from the downloaded community zip

**Comment by ilikerainproton191** [comment]
> > As a temporary fix I migrated over to community edition, it seems to work fine. As I did not have license im not sure if your branding carries over or not.
> 
> How did you manage to do that ? I do not have a license too but I have the timeout as well.

If you are lazy to apply entire #2148, open /rainloop/v/1.16.0/app/libraries/RainLoop/Providers/Prem.php, and replace [this code](https://github.com/RainLoop/rainloop-webmail/blob/master/rainloop/v/0.0.0/app/libraries/RainLoop/Providers/Prem.php#L138-L223) with `return 'EXPIRED:1898625600';` if you want to have access to premium feature, or with `return 'NO';` if you don't want it.

If you insist on switching to Community edition regardless, you should actually care to search for word "community" in this repository ([here](https://github.com/RainLoop/rainloop-webmail/search?q=community)), you will notice that the only differences are the absence of Prem.php and different JS files. So you just remove this file and replace the JS files with the ones in Community edition archive.

**Comment by rainarr1** [comment]
Missed a few files, [here](https://i.imgur.com/Yso2KVC.png) is an actual diff.

**Comment by amarc** [comment]
Does anyone know how to contact guy other than support@ email ? We paid license $300 less than 2 months ago and now we can't even know if developer is alive or not..

**Comment by passpro22** [comment]
> Does anyone know how to contact guy other than support@ email ? We paid license $300 less than 2 months ago and now we can't even know if developer is alive or not..

Treat it as wasted money and don't help copyright holders get more rich. Assuming they are even copyright holders (see issues mentioned earlier).

You should have taken a look at what you are paying for.

IMHO $300 is insane amount for just branding and simple updater, which can be done by another programmer for much cheaper.

**Comment by benchonaut** [comment]
( unfortunately snappymail that is a more-or-less rewrite of rainloop has the appdata-stuck problem as well - at least with 2 apaches in a row )
